__tests__/app.test.js

```diff
@@ -17,76 +17,114 @@ afterAll(() => {
+        return request(app)
             .get("/api/articles/4")
             .expect(200)
-            .then(({body}) => { 
+            .then(({ body }) => {
                 expect(body.article).toMatchObject({
                     author: expect.any(String),
                     title: expect.any(String),
-                    article_id: expect.any(Number), 
+                    article_id: expect.any(Number),
                     body: expect.any(String),
                     topic: expect.any(String),
                     created_at: expect.any(String),
                     votes: expect.any(Number),
                     article_img_url: expect.any(String)
                 })
             })
-        })
-        test("status: 400, responds with an error message when passed a bad article_id", () => {
-            return request(app)
+    })
+    test("status: 400, responds with an error message when passed a bad article_id", () => {
+        return request(app)
             .get("/api/articles/notAnId")
             .expect(400)
-            .then(({ body }) => { 
+            .then(({ body }) => {
                 expect(body.msg).toBe("Invalid input")
             })
-        })
-    })
\ No newline at end of file
+    })
+    test("status: 404, responds with an error message when passed a valid article_id that doesn't exist in the database", () => {
+        return request(app)
+            .get("/api/articles/9999")
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("Not found")
+            })
+    })
+})
+
+describe("/api/articles", () => {
+    test("status: 200, responds with an array of article objects with the properties author, title, article_id, topic, created_at, votes, article_img_url and comment_count", () => {
+        return request(app)
+            .get("/api/articles")
+            .expect(200)
+            .then((({ body: { articles } }) => {
+                expect(articles.length).toBe(13)
+                articles.forEach((article) => {
+                    expect(article).toMatchObject({
+                        author: expect.any(String),
+                        title: expect.any(String),
+                        article_id: expect.any(Number),
+                        topic: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number),
+                        article_img_url: expect.any(String),
+                        comment_count: expect.any(Number)
+                    })
+                })
+            }))
+    })
+    test("responds with articles sorted by date in descending order", () => { 
+        return request(app)
+        .get("/api/articles")
+        .then((({body: { articles} }) => {
+            expect(articles).toBeSortedBy("created_at", {descending: true})
+        }))
+    })
+})
```

Comment:
What about testing different queries for sort by and order? Since this is something that the user can input, this is a likely place for error to pop up...
Worth thinking about what we should send back for cases such as `/api/articles?sort_by=banana`

And also just testing cases where it's used correctly would be great